[kealib] build failure when gdal is installed first

### Operating system

Windows/Linux/OSX static triplets

### Compiler

_No response_

### Steps to reproduce the behavior

```Shell
.\vcpkg.exe install hdf5[core,cpp,szip,tools,zlib]:x64-windows-static gdal[core,curl,default-features,expat,geos,gif,hdf5,iconv,jpeg,lerc,libkml,libspatialite,libxml2,lzma,netcdf,openjpeg,openssl,pcre2,png,postgresql,qhull,recommended-features,sqlite3,webp,zstd]:x64-windows-static
.\vcpkg.exe install kealib:x64-windows-static --binarysource clear
```

As discovered in https://dev.azure.com/vcpkg/public/_build/results?buildId=91885&view=results
```


### Failure logs

```
Installing 1/1 kealib:x64-windows-static...
Building kealib:x64-windows-static...
-- Using cached ubarsc-kealib-4984102cf5867105a28ae754689566217309d120.tar.gz.
-- Cleaning sources at C:/Dev/vcpkg/buildtrees/kealib/src/217309d120-809ed366ba.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Dev/vcpkg-downloads/ubarsc-kealib-4984102cf5867105a28ae754689566217309d120.tar.gz
-- Applying patch hdf5_include.patch
-- Using source at C:/Dev/vcpkg/buildtrees/kealib/src/217309d120-809ed366ba.clean
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Dev/vcpkg-downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j33
    Working Directory: C:/Dev/vcpkg/buildtrees/kealib/x64-windows-static-dbg
    See logs for more information:
      C:\Dev\vcpkg\buildtrees\kealib\install-x64-windows-static-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/kealib/portfile.cmake:16 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building kealib:x64-windows-static failed with: BUILD_FAILED
```

[install-x64-windows-static-dbg-out.log](https://github.com/microsoft/vcpkg/files/12101323/install-x64-windows-static-dbg-out.log)


### Additional context

I'm investigating

--- a/ports/kealib/portfile.cmake
+++ b/ports/kealib/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DLIBKEA_WITH_GDAL=OFF
+        -DCMAKE_DISABLE_FIND_PACKAGE_GDAL=ON
 )
 
 vcpkg_cmake_install()

--- a/ports/kealib/vcpkg.json
+++ b/ports/kealib/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "kealib",
   "version": "1.4.14",
+  "port-version": 1,
   "description": "KEALib provides an implementation of the GDAL data model using HDF5.",
   "homepage": "https://github.com/ubarsc/kealib",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3566,7 +3566,7 @@
     },
     "kealib": {
       "baseline": "1.4.14",
-      "port-version": 0
+      "port-version": 1
     },
     "kenlm": {
       "baseline": "20230531",

--- a/versions/k-/kealib.json
+++ b/versions/k-/kealib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d6136ef42467c0204cd6a082a272c16ab955caa3",
+      "version": "1.4.14",
+      "port-version": 1
+    },
+    {
       "git-tree": "11878c71d1418c0bdcc380e7ebf8e90f9e563260",
       "version": "1.4.14",
       "port-version": 0

